lib/puppet/network/http/api/v3/indirected_routes.rb

```diff
@@ -88,6 +87,8 @@ def uri2indirection(http_method, uri, params)
       params[:environment] = configured_environment
     end
 
+    check_authorization(method, "#{self.class.url_prefix}/#{indirection_name}/#{key}", params)
+
```

Comment:
This doesn't appear to be a new issue since the same code path appears to be in place on "stable" -- https://github.com/puppetlabs/puppet/blob/stable/lib/puppet/network/http/api/v1.rb#L68 -- but I agree this would be a good idea.  The information disclosure we'd allow for otherwise seems inappropriate.  I'll pursue that if there are no other objections...